crates/core/src/kernel/models/actions.rs

```diff
@@ -6,118 +6,132 @@ use delta_kernel::schema::{DataType, StructField};
 use delta_kernel::table_features::{ReaderFeature, WriterFeature};
 use maplit::hashset;
 use serde::{Deserialize, Serialize};
-use tracing::warn;
 
 use crate::kernel::{error::Error, DeltaResult};
 use crate::kernel::{StructType, StructTypeExt};
 use crate::TableProperty;
 
-/// Defines a file format used in table
-#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
-pub struct Format {
-    /// Name of the encoding for files in this table
-    pub provider: String,
-    /// A map containing configuration options for the format
-    pub options: HashMap<String, Option<String>>,
+pub use delta_kernel::actions::{Metadata, Protocol};
+
+pub fn new_metadata(
+    schema: &StructType,
+    partition_columns: impl IntoIterator<Item = impl ToString>,
+    configuration: impl IntoIterator<Item = (impl ToString, impl ToString)>,
+) -> DeltaResult<Metadata> {
+    let value = serde_json::json!({
+        "id": uuid::Uuid::new_v4().to_string(),
+        "name": None::<String>,
+        "description": None::<String>,
+        "format": { "provider": "parquet", "options": {} },
+        "schemaString": serde_json::to_string(schema)?,
+        "partitionColumns": partition_columns.into_iter().map(|c| c.to_string()).collect::<Vec<_>>(),
+        "configuration": configuration.into_iter().map(|(k, v)| (k.to_string(), v.to_string())).collect::<HashMap<_, _>>(),
+        "createdTime": chrono::Utc::now().timestamp_millis(),
+    });
```

Comment:
eeeeep! you weren't kidding about some messiness! As far as my sleep-deprived brain can tell this is because the `Matadata` inside of kernel is not exposing any constructors outward? 

Assuming that's the case, would it be viable to add a builder API into kernel for these actions? I imagine this is a useful capability for transaction log utilities and other applications that might want to use kernel